File: src/main/java/org/myfintech/payment/repository/PaymentTrackingRepository.java
package org.myfintech.payment.repository;

import java.util.Optional;

import org.myfintech.payment.entity.PaymentTracking;

public interface PaymentTrackingRepository extends BaseRepository<PaymentTracking, Long> {
	Optional<PaymentTracking> findByTrackingNumber(String trackingNumber);
	
	
}

File: src/main/kotlin/org/myfintech/payment/repository/PaymentTrackingRepository.kt
package org.myfintech.payment.repository

import org.myfintech.payment.entity.PaymentTracking
import java.util.*

interface PaymentTrackingRepository : BaseRepository<PaymentTracking, Long> {
    fun findByTrackingNumber(trackingNumber: String?): Optional<PaymentTracking>
}
